src/Providers/JWT/Provider.php

```diff
@@ -42,6 +43,10 @@ abstract class Provider
      */
     public function __construct($secret, $algo, array $keys)
     {
+        if(is_null($secret) && (is_null($keys['public']) || is_null($keys['private']))) {
+            throw new SecretMissingException();
+        }
```

Comment:
The only improvement I an think would be to include a nice message, pointing out _which_ secret is missing specifically.